gateway-service-api/src/main/proto/org/hypertrace/gateway/service/v1/log_event.proto

```diff
@@ -0,0 +1,22 @@
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "org.hypertrace.gateway.service.v1.log.events";
+
+package org.hypertrace.gateway.service.v1.log.event;
+
+import "org/hypertrace/gateway/service/v1/gateway_query.proto";
+
+message LogEventRequest {
+  org.hypertrace.gateway.service.v1.common.Filter filter = 1;
+  repeated org.hypertrace.gateway.service.v1.common.Expression selection = 2;
+}
+
+message LogEvent {
+  map<string, org.hypertrace.gateway.service.v1.common.Value> attributes = 1;
+}
+
+message LogEventResponse {
+  repeated LogEvent logEvents = 1;
+  int32 total = 2;
```

Comment:
My understanding is this needs to be fetched separately, so we've kind of retrofitted an "includeTotal" flag on other APIs. Since this is a new one, it may be worth looking at those and if it makes sense, splitting fetching total into its own API.

Reply:
Why does `total` needs to be fetched separately? and if that is added, should the existing api also return `total`?

Reply:
It was a perf optimization. In many contexts (for example, showing the log events in the sequence diagram), we don't need the total - but under the hood, the gateway service had to make a separate query (which I believe was sequential, although that could be fixed) to fetch it and it slowed down the queries that didn't need it. By keeping it separate, we can fetch it in parallel if needed and leave it be if it's not requested. @skjindal93 may have more context.

Reply:
> Why does `total` needs to be fetched separately? and if that is added, should the existing api also return `total`?

I think there are 2 questions here

1. Why does `total` needs to be fetched separately?

If the request has support for **limit** and **offset**, which usually gets pushed down to the data store, you cannot count number of log events in the response and call it as **total** (since the count of log events would just be bounded by the **limit**)
A separate request needs to be made to get the **total** value, by just passing the filters sent in the request (_And, that's why I personally prefer to have a separate total request API entirely_)

2. What if you don't need total in the response?
That's the perf optimization which @aaron-steinfeld was mentioning about. Since, in order to fetch **total**, you need to make a separate request, but if the consumer does not need it, we should skip fetching the **total** entirely (_by either having a boolean flag like this https://github.com/hypertrace/gateway-service/blob/main/gateway-service-api/src/main/proto/org/hypertrace/gateway/service/v1/entities.proto#L50 or a separate total request API_)

Reply:
Actually, since I didn't add support for limit and order, calculating total was trivial. But, yeah with limit and order it needs another query. 
For this change, I will remove `total` from the api.